### ISSUE TITLE
Question: How to validate the same Jira issue is both in the branch name and at the start of commit message

### ISSUE THREAD

**Issue by tlvu** [tomasbjerre/simple-bitbucket-commit-checker#87]
Hi,

The use-case is to prevent use from typing the wrong Jira issue into the commit message.

We typically create the branch from Jira so there is no way to get the branch name wrong and the Jira issue will also be part of the branch name.

So if we can check that the same Jira issue that is in the branch name is also in the commit message, we are sure it is good.

This exclude branch master and develop of course. This is for a bugfix/feature/hotfix/release or other custom type of branches.

Thanks

**Comment by tomasbjerre** [comment]
You cannot validate that it is the same jira. I guess you figured out how to validate that there is "a" jira.

**Comment by tlvu** [comment]
Yes I am able to validate there is "a" jira with a regex on a commit.  What I am after is able to validate both the commit and the branch name at the same time and check they contain the same jira.

**Comment by gsylvie** [comment]
Neat idea.  If you can wait 3-4 weeks I'll implement this in _my_ free JIRA checker (it's called "[Control Freak](https://marketplace.atlassian.com/apps/1217635/control-freak-for-bitbucket-server?hosting=server&tab=overview)").

Everyone seems to have their own free Bitbucket JIRA checker these days....  ;-)

@tomasbjerre - I hope you don't mind me posting here.  Ping me at sylvie@bit-booster.com any time.

**Comment by tomasbjerre** [comment]
No worries! As long as your stuff is free, I don't mind at all =)
